contrib/cirrus/setup.sh

```diff
@@ -17,4 +17,26 @@ trap "complete_setup" EXIT
 msg "************************************************************"
 msg "Setting up runtime environment"
 msg "************************************************************"
+
 show_env_vars
+
+if [[ "$CIRRUS_TASK_NAME" == "build_cross" ]]; then
+	msg "Installing podman"
+	dnf -y install podman
+	# Setup short-name for rustembedded/cross
+	# TODO: We can move this to quay.io if we reach rate-limits, hopefully that's not gonna happen for netavark
+	echo '  "rustembedded/cross" = "docker.io/rustembedded/cross"'  >> /etc/containers/registries.conf.d/000-shortnames.conf
+	msg "Installing gcc cross-compile packages"
+	err_retry 8 1000 "" dnf install -y \
+	    qemu-user-static \
+	    'gcc-*-linux-gnu' \
+	    'gcc-c++-*-linux-gnu'
+	# This is required by rustup-init (it will complain if any tools are in $PATH)
+	rust_version=$(rpm -q --qf '%{v}' rust)
+	msg "Replacing Fedora native rust toolchain with upstream '$rust_version'"
+	dnf erase -y cargo rust rust-std-static
+	msg "Initializing upstream rust environment."
+	mkdir -p $CARGO_HOME/bin/
+	cd $CARGO_HOME/bin/
+	curl https://sh.rustup.rs -sSf | sh -s -- -y
```

Comment:
FYI: I'm not against abandoning the Fedora-rust toolchain completely, and simply installing upstream rust at VM image build time.

Reply:
I think this is being discussed in above comment.

Reply:
Let's just do this, it will make life much easier.